eval/NEESEvaluator.h

```diff
@@ -0,0 +1,95 @@
+#pragma once
+
+#include <gtsam/navigation/ImuFactor.h>
+#include <gtsam/navigation/CombinedImuFactor.h>
+#include <gtsam/navigation/PreintegrationCombinedParams.h>
+#include <optional>
+#include <vector>
+
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file    NEESEvaluator.h
+ * @brief   Normalized Estimation Error Squared (NEES) Metrics Evaluator
+ * @author  Alec Kain
+ */
+
+namespace gtsam {
+
+/**
+ * @brief Evaluator for Normalized Estimation Error Squared (NEES) metrics
+ * 
+ * This class evaluates the NEES metric for IMU preintegration over different time windows.
+ * It uses the EuRoC MAV dataset for ground truth and IMU measurements.
+ * 
+ * The NEES metric helps assess the consistency of the estimator by comparing
+ * the predicted state with ground truth, normalized by the estimator's covariance.
+ */
+
+class NEESEvaluator {
+public:
+    struct NoiseParams {
+        double sigma_gyro;
+        double sigma_acc;
+        double sigma_gyro_bias;
+        double sigma_acc_bias;
+    };
+
+    struct GTData {
+        std::vector<NavState> nav_states;
```

Comment:
If any of these vectors are supposed to be the same length, this is a design flaw, because I can break it :-) consider in that case vector<SmallStruct>. Added benefit: if measurements and states are at deifferent frequencies, I’ll know because then they’ll live in different vectors